src/clusterfuzz/_internal/bot/tasks/trials.py

```diff
@@ -61,17 +68,36 @@ def __init__(self):
       for config in trials_config:
         if config['app_name'] != app_name:
           continue
-        self.trials[config['app_args']] = config['probability']
+        self.trials[config['app_args']] = AppArgs(config['probability'],
+                                                  config.get('contradicts', []))
     except Exception as e:
       logs.log_warn('Unable to parse config file: %s' % str(e))
       return
 
-  def setup_additional_args_for_app(self):
+  def doesnt_asymmetrically_contradicts(self, app_args, existing_contradictions,
+                                        trial_args):
```

Comment:
Please add a docstring for this function.

Reply:
I have moved the conditions into the loop with comments instead